NachoSpigot-Server/src/main/java/net/minecraft/server/EntityLiving.java

```diff
@@ -1789,7 +1790,28 @@ public void receive(Entity entity, int i) {
     }
 
     public boolean hasLineOfSight(Entity entity) {
-        return this.world.rayTrace(new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ), new Vec3D(entity.locX, entity.locY + (double) entity.getHeadHeight(), entity.locZ)) == null;
+        if (entity instanceof EntityPlayer && NachoConfig.enableImprovedHitReg) {
+            // Head height is 1,5725
+            // Split it into three to get a more accurate line of sight -> 0.52416667
+
+            double parts = entity.getHeadHeight() / 3;
```

Comment:
It's kinda iffy dividing by 3 and calling ray trace, as that's one of the most expensive calls you can make, is there any other solutions here?

Reply:
I personally don't know of any other ways to do it